OSPSuite.Presentation.Importer/Services/IImporter.cs

```diff
@@ -0,0 +1,51 @@
+﻿using OSPSuite.Presentation.Importer.Core;
+using System.Collections.Generic;
+using System.Linq;
+using IoC = OSPSuite.Utility.Container.IContainer;
+
+namespace OSPSuite.Presentation.Importer.Services
+{
+   public interface IImporter
+   {
+      IDataSourceFile LoadFile(string path);
+      IDataSource ImportFromFile(IEnumerable<IDataSheet> sheets);
+      IList<IDataFormat> AvailableFormats(IUnformattedData data);
+   }
+
+   public class Importer : IImporter
+   {
+      private readonly IoC container;
+      private readonly IDataSourceFileParser parser;
+
+      public Importer(IoC container)
+      {
+         this.container = container;
+         this.parser = container.Resolve<IDataSourceFileParser>();
+      }
+
+      public IList<IDataFormat> AvailableFormats(IUnformattedData data)
+      {
+         return GetType().Assembly.GetTypes()
```

Comment:
containerhas a resolveAll or something like this that you can use
You can even use the IRepository<IDataFormat> in constructor which will inject a generic service doing exactly this

Reply:
Done :)

Reply:
Done